### PR TITLE
Add documentation for Unit Hint

### PR DESCRIPTION
**Pull request by andrewfg**
See https://github.com/openhab/openhab-core/pull/4079

Signed-off-by: Andrew Fiddian-Green <software@whitebear.ch>

### CODE REVIEW

**Review by mherwege — APPROVED** [review]
LGTM Thank you.

File: developers/bindings/thing-xml.md [hunk 1]
```diff
@@ -93,8 +94,16 @@ The following XML snippet shows a thing type definition with 2 channels and one
...
 </channel-type>
+<channel-type id="humidity" advanced="false">
+    <item-type unitHint="%">Number:Dimensionless</item-type>
+    <label>Humidity</label>
+    <state readOnly="true" pattern="%.1f %%"></state>
+</channel-type>
 ```
 
+The `item-type` element gives the [item type](../../configuration/items.md#type) to be used when a linked item is created.
+If the `item-type` is a `Number:<dimension>` then a `unitHint` attribute may be provided to suggest the measurement unit to be used when a linked item is created.
```
[the remark this answers is not in the document]
Comment (andrewfg) [reply developers/bindings/thing-xml.md:105]: On grammatical grounds I would generally disagree with your proposed comma. But will add it anyway for an easy life.

[the remark this answers is not in the document]
Comment (david-pace) [reply developers/bindings/thing-xml.md:105]: I know that this comma is not intuitive, but here are some sources:
* https://www.constant-content.com/content-writing-service/2007/12/if-then-statements-require-commas/
* https://www.grammarly.com/blog/conditional-sentences/

I also checked the sentence `If the item type is a number then a unit hint attribute may be provided.` with
* https://www.grammarcheck.net/editor/
* https://quillbot.com/grammar-check
* https://languagetool.org/

and all checkers demand a comma after `number,`.

**Review by david-pace — APPROVED** [review]

File: developers/bindings/thing-xml.md [hunk 2]
```diff
@@ -85,6 +85,7 @@ The following XML snippet shows a thing type definition with 2 channels and one
     <channels>
         <channel id="switch" typeId="powerSwitch" />
         <channel id="temperature" typeId="setpointTemperature" />
+        <channel id="roomHumidity" typeId="humidity" />
```
Comment (jlaur) [inline developers/bindings/thing-xml.md:88]: Naming convention:
```suggestion
        <channel id="room-humidity" typeId="humidity" />
```

If you want to extend scope slightly, you could also correct the two channel type ids to comply with the now established naming convention.

File: developers/bindings/thing-xml.md [hunk 3]
```diff
@@ -93,8 +94,16 @@ The following XML snippet shows a thing type definition with 2 channels and one
     <category>Temperature</category>
     <state min="12" max="30" step="0.5" pattern="%.1f °C" readOnly="false" />
 </channel-type>
+<channel-type id="humidity" advanced="false">
```
Comment (jlaur) [inline developers/bindings/thing-xml.md:97]:
```suggestion
<channel-type id="humidity">
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Two minor comments from my side.

File: developers/bindings/thing-xml.md (hunk 2 above)
Comment (andrewfg) [reply developers/bindings/thing-xml.md:88]: Umm. On this page, there are several more cases of either channel id or channel type id being camel case. So to be fully consistent I think we should change all instances of both channel id and channel type id from camelCase to dash-separated .. or??

Comment (jlaur) [reply developers/bindings/thing-xml.md:88]: Yes, that would be even better. We don't want to encourage using camelCase only to be corrected when providing a PR.

**Comment by stefan-hoehn** [comment]
@jlaur Can you have another look after the changes?

**Review by jlaur — APPROVED** [review]
LGTM

**Review by stefan-hoehn — APPROVED** [review]

**Pull request merged**